### PR TITLE
Fixup publish images

### PR DESCRIPTION
**Pull request by elmiko**
**What this PR does / why we need it**:

this change adds some fixes to the image publishing tool to make it more resilient to errors and improve some of the behavior around pushing images.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

the tool can be built by running `go build publish-kubemark-images.go` in the `hack/tools` directory

**Release notes**:

```release-note
```

### CODE REVIEW

File: hack/tools/publish-kubemark-images.go [hunk 1]
```diff
@@ -152,7 +153,7 @@ func (p *PublishImages) imageExists(v *semver.Version) (bool, error) {
 	cmd.Stderr = os.Stderr
 	if err := cmd.Run(); err != nil {
 		if exitError, ok := err.(*exec.ExitError); ok {
-			if exitError.ExitCode() == 1 {
+			if exitError.ExitCode() == 1 || exitError.ExitCode() == 125 {
```
Comment (chrischdi) [inline hack/tools/publish-kubemark-images.go:156]: Maybe we could add a comment about what this exit codes are :-)

**Review by chrischdi — COMMENTED** [review]
Change makes sense to me

/lgtm

**Review by fabriziopandini — COMMENTED** [review]
Getting late to the party bust asking some questions, mostly because of me learning this project...
Why are we using k8s.gcr.io/kube-proxy as repository instead of staging?
Is there a specific reason why this provider is using a golang program for build and release, while most of the other providers that I know are using plain makefiles for this task?

**Comment by fabriziopandini** [comment]
Also, why are we using sigs.k8s.io/controller-tools v0.4.1-0.20201002000720-57250aac17f6 instead of a tagged release (e.g in CAPI we use v0.10.0 in main now)

**Comment by elmiko** [comment]
hey @fabriziopandini , yes, excellent questions all of them!

this specific application is a remnant left over from the early days of the project. we use it for building the actual kubemark image which is deployed on the cluster (not the capi kubemark controller). i agree that this tool could use some maintenance and cleanup to the way we use it. so far it has been a helper to ensure we can keep using kubemark in the cluster, but i am not opposed to the cleanups you are talking about.

i would also like to see our creation of the kubemark images automated, but given the resources it takes to do a full build this has not materialized yet.  i've mainly been making the images and pushing to my personal registry, https://quay.io/repository/elmiko/kubemark?tab=tags&tag=latest.

i will circle back to these suggestions and create some issues to capture what should be done.  thanks!

**Pull request merged**